test/subdomainregistrar/TestRentalSubdomainRegistrar.js

```diff
@@ -0,0 +1,778 @@
+        namehash('subname.test.eth'),
+      )
+      await SubdomainRegistrar2.renew(node, 'subname', 86400)
+      const [, , expiry2] = await NameWrapper.getData(
+        namehash('subname.test.eth'),
+      )
+
+      expect(expiry2.toNumber()).to.equal(expiry.toNumber() + 86400)
+    })
+
+    it('should revert if expiry is greater than parent', async () => {
+      await expect(
+        SubdomainRegistrar2.renew(node, 'subname', 84600 * 3),
+      ).to.be.revertedWith(`DurationTooLong("${namehash('test.eth')}")`)
+    })
+
+    it('should emit an event on renewal', async () => {
+      const tx = await SubdomainRegistrar2.renew(node, 'subname', 500)
+      const [, , expiry] = await NameWrapper.getData(
+        namehash('subname.test.eth'),
+      )
+      await expect(tx)
+        .to.emit(SubdomainRegistrar, 'NameRenewed')
+        .withArgs(namehash('subname.test.eth'), expiry)
+    })
+  })
+
+  describe('register Subnames with records', () => {
+    it('should allow a subname to be registered with records', async () => {
+      const node = namehash('test.eth')
+      await BaseRegistrar.register(labelhash('test'), account, 86400 * 2)
+      await BaseRegistrar.setApprovalForAll(NameWrapper.address, true)
+      await NameWrapper.wrapETH2LD(
+        'test',
+        account,
+        CANNOT_UNWRAP,
+        EMPTY_ADDRESS,
+      )
+      expect(await NameWrapper.ownerOf(node)).to.equal(account)
+      await SubdomainRegistrar.setupDomain(
+        node,
+        RentalPricerFree.address,
+        EMPTY_ADDRESS,
+        true,
+      )
+      await NameWrapper.setApprovalForAll(SubdomainRegistrar.address, true)
+      await SubdomainRegistrar2.register(
+        node,
+        'subname',
+        account2,
+        PublicResolver.address,
+        0,
+        86400,
+        [
+          PublicResolver.interface.encodeFunctionData(
+            'setAddr(bytes32,address)',
+            [subNode, account2],
+          ),
+        ],
+      )
+
+      expect(await NameWrapper.ownerOf(namehash('subname.test.eth'))).to.equal(
+        account2,
+      )
+      expect(await PublicResolver['addr(bytes32)'](subNode)).to.equal(account2)
+    })
+  })
+
+  describe('batchRegister()', () => {
+    it('should allow subnames to be batch registered with records', async () => {
+      const node = namehash('test.eth')
+      await BaseRegistrar.register(labelhash('test'), account, 86400 * 2)
+      await BaseRegistrar.setApprovalForAll(NameWrapper.address, true)
+      await NameWrapper.wrapETH2LD(
+        'test',
+        account,
+        CANNOT_UNWRAP,
+        EMPTY_ADDRESS,
+      )
+      expect(await NameWrapper.ownerOf(node)).to.equal(account)
+      await SubdomainRegistrar.setupDomain(
+        node,
+        RentalPricerFree.address,
+        account,
+        true,
+      )
+      await NameWrapper.setApprovalForAll(SubdomainRegistrar.address, true)
+      await Erc20WithAccount2.approve(
```

Comment:
Do you need to approve? Assuming you are testing registration without fee as the next one is explicitly testing with fee